Assets/Scripts/JobSystem/DragNDrop.cs

```diff
@@ -49,10 +53,16 @@ public void OnEndDrag(PointerEventData eventData)
 
         canvasGroup.alpha = JobConstants.dropAlpha; 
 
-        // Return to starting position if not dropped in schedule 
-        if (gameObject.transform.parent != scheduleContainer.transform)
+        // Return to starting position if not dropped in a slot  
+        if (!jobsUI.IsInsideSlot(gameObject))
         {
             gameObject.transform.parent = availableJobsContainer.transform;
+
+            // If the job was previously scheduled, update the schedule accordingly 
+            if (job.isScheduled)
+            {
+                onUnscheduleJob?.Invoke(job);

```

Comment:
the name `onUnscheduleJob` seems unintuitive here

Reply:
Well this logic fires if the player drags the job out of the schedule. In this scenario, it was already scheduled (already inside the schedule container), then it becomes unscheduled (goes back to the available jobs container). I think it's okay personally. But if you have a better name, I'm all ears. 

The comment is a tad misleading though. Will change that.